lib/koans/lists.ex

```diff
@@ -1,7 +1,103 @@
 defmodule Lists do
   use Koans
 
-  koan "We shall contemplate truth by testing reality, via equality" do
-    assert List.first([1, 2, 3]) == 2
+  koan "We can see what is ahead" do
+    assert List.first([1, 2, 3]) == 1
+  end
+
+  koan "Checking what's trailing is also simple" do
+    assert List.last([1, 2, 3]) == 3
+  end
+
+  koan "Diversity is embraced" do
+    result = [1, 2] ++ [:a, "b"]
+    assert result == [1, 2, :a, "b"]
+  end
+
+  koan "Things can evolve" do
+    result = [1, 2, 3] -- [3]
+    assert result == [1, 2]
+  end
+
+  koan "Evolution can have different forms" do
+    result = List.delete([1, 2, 2, 3], 2)
+    assert result == [1, 2, 3]
+  end
+
+  koan "Precision is also valued" do
+    result = List.delete_at([1, 2, 3], 1)
+    assert result == [1, 3]
+  end
+
+  koan "Replication is also possible" do
+    result = List.duplicate("life", 3)
+    assert result == ["life", "life", "life"]
+  end
+
+  koan "Sometimes levelling the playing field is desired" do
+    result = List.flatten([1, [2, 3], 4, [5]])
+    assert result == [1, 2, 3, 4, 5]
+  end
+
+  koan "Same rules apply to new members that arrive late" do
+    result = List.flatten([1, [2, 3]], [4])
+    assert result == [1, 2, 3, 4]
+  end
+
+  koan "You can fold everything into one" do
+    result = List.foldl([1, 2, 3, 4], 0, fn(x, acc) -> x - acc end)
+    assert result == 2
+  end
+
+  koan "Be careful of the fold" do
+    result = List.foldr([1, 2, 3, 4], 0, fn(x, acc) -> x - acc end)
+    assert result == -2
+  end
+
+  koan "Order can also be specified for new members" do
+    result = List.insert_at([1, 2, 3], 1, 4)
+    assert result == [1, 4, 2, 3]
+  end
+
+  koan "We can replace things at specified positions" do
+    result = List.replace_at([1, 2, 3], 0, 10)
+    assert result == [10, 2, 3]
+  end
+
+  koan "Replacing something which is not" do
+    result = List.replace_at([1, 2, 3], 10, 0)
+    assert result == [1, 2, 3]
+  end
+
+  koan "Order is bound by nature's laws" do
+    result_one = List.insert_at([1, 2, 3], 10, 4)
+    result_two = List.insert_at([1, 2, 3], -1, 4)
+
+    assert result_one == [1, 2, 3, 4]
+    assert result_two == [1, 2, 3, 4]
+  end
+
+  koan "Updating an existing element can be more complex" do
```

Comment:
I was thinking that maybe functions should come after lists. In that case, this kata could be confusing to users :)